Code cleanup: Allow internal code to call "timegm" on Windows

PR #3544 tried a change but hit:
..\..\tests\crypto\crl_tests.c(20): error C2220: warning treated as error - no 'object' file generated
..\..\tests\crypto\crl_tests.c(20): warning C4005: 'timegm': macro redefinition
C:\Users\oeadmin\workspace\pipelines\Azure-Windows\include\openenclave/internal/datetime.h(21): note: see previous definition of 'timegm'

Need to investigate how to clean this up, to avoid platform-specific ifdefs in .c files

As noted in PR #3544 comment:

The error was
```
..\..\tests\crypto\crl_tests.c(20): error C2220: warning treated as error - no 'object' file generated
..\..\tests\crypto\crl_tests.c(20): warning C4005: 'timegm': macro redefinition
C:\Users\oeadmin\workspace\pipelines\Azure-Windows\include\openenclave/internal/datetime.h(21): note: see previous definition of 'timegm'
```
and that's because `....\tests\crypto\crl_tests.c` already does the same thing as what I said to put in `datetime.h`.
All that's needed to clean this up is to delete `....\tests\crypto\crl_tests.c` lines 19-21.

--- a/common/datetime.c
+++ b/common/datetime.c
@@ -285,11 +285,7 @@ oe_result_t oe_datetime_to_time_t(const oe_datetime_t* datetime, time_t* value)
     timeinfo.tm_min = (int)datetime->minutes;
     timeinfo.tm_sec = (int)datetime->seconds;
 
-#ifdef _WIN32
-    tmp = _mkgmtime(&timeinfo);
-#else
     tmp = timegm(&timeinfo);
-#endif
 
     *value = tmp;
 

--- a/include/openenclave/internal/datetime.h
+++ b/include/openenclave/internal/datetime.h
@@ -18,6 +18,7 @@ OE_EXTERNC_BEGIN
 
 #ifdef _WIN32
 #define gmtime_r(now, timeinfo) gmtime_s(timeinfo, now)
+#define timegm(tm) _mkgmtime(tm)
 #endif
 
 /**

--- a/tests/crypto/crl_tests.c
+++ b/tests/crypto/crl_tests.c
@@ -16,9 +16,6 @@
 #include <time.h>
 #include "readfile.h"
 #include "tests.h"
-#if defined(_WIN32)
-#define timegm _mkgmtime
-#endif
 
 #define ACCEPTABLE_ERROR_IN_SECONDS 60
 

